templates/node/index.d.ts.twig

```diff
@@ -115,10 +115,19 @@ declare module "{{ language.params.npmPackage|caseDash }}" {
 {% endfor %}
   }
 
-  export class AppwriteException extends Error {
-    public code: number | null;
-    public response: string | null;
-    constructor(message: string, code?: number, response?: string);
+  export type {{spec.title | caseUcfirst}}ExceptionResponse = {
+        message: string;
+        code: number;
+        type: string;
+        version: string;
+  };
+
+  export class {{spec.title | caseUcfirst}}Exception extends Error {
+    public code: number;
+    public type: string;
+    public response: {{spec.title | caseUcfirst}}ExceptionResponse;
+
+    constructor(message: string, code?: number, type?: string, response?: {{spec.title | caseUcfirst}}ExceptionResponse);
```

Comment:
I see the signature of this function call has changed. Is this expected?

Reply:
![CleanShot 2022-09-21 at 16 13 05](https://user-images.githubusercontent.com/19310830/191529482-b855ad93-9942-444d-a0f4-1b6e8ba1d73d.png)

`d.ts`file now follows the same structure we have on the actual JS file. We just fixed it to be cirrect.